addons/sourcemod/scripting/szf/pickupweapons.sp

```diff
@@ -541,33 +539,50 @@ stock void SetRandomWeapon(int iEntity, eWeaponsRarity nRarity)
 	eWeapon wep;
 	array.GetArray(iRandom, wep);
 	
-	SetWeaponModel(iEntity, wep.sModel);
+	float flOffsetOrigin[3];
+	float flOffsetAngles[3];
+	Weapons_GetOffsets(wep, flOffsetOrigin, flOffsetAngles);
+	
+	SetWeaponModel(iEntity, wep);
 	
-	if (wep.color[0] + wep.color[1] + wep.color[2] > 0)
+	if (wep.iColor[0] + wep.iColor[1] + wep.iColor[2] > 0)
 	{
 		SetEntityRenderMode(iEntity, RENDER_TRANSCOLOR);
-		SetEntityRenderColor(iEntity, wep.color[0], wep.color[1], wep.color[2], 255);
+		SetEntityRenderColor(iEntity, wep.iColor[0], wep.iColor[1], wep.iColor[2], 255);
 	}
 	
 	delete array;
 }
 
-stock void SetWeaponModel(int iEntity, char[] strModel)
+stock void SetWeaponModel(int iEntity, eWeapon wep)
 {
+	float flOffsetOrigin[3];
+	float flOffsetAngles[3];
+	Weapons_GetOffsets(wep, flOffsetOrigin, flOffsetAngles);
+	
 	char strOldModel[256];
 	GetEntityModel(iEntity, strOldModel, sizeof(strOldModel));
 	
-	// Because sniper wearable have a really offplace origin prop, we have to move entity to a more reasonable spot
+	// Offsets
+	float vecOrigin[3];
+	GetEntPropVector(iEntity, Prop_Send, "m_vecOrigin", vecOrigin);
+	vecOrigin[0] += flOffsetOrigin[0];
+	vecOrigin[1] += flOffsetOrigin[1];
+	vecOrigin[2] += flOffsetOrigin[2];
+	
+	float vecAngles[3];
+	GetEntPropVector(iEntity, Prop_Send, "m_angRotation", vecAngles);
+	vecAngles[0] += flOffsetAngles[0];
+	vecAngles[1] += flOffsetAngles[1];
+	vecAngles[2] += flOffsetAngles[2];
 	
+	TeleportEntity(iEntity, vecOrigin, vecAngles, NULL_VECTOR);
```

Comment:
If your your going to make `offset_origin`, `offset_angles` and `scale` only take effect when setting model, and not taking consideration when model get replaced to something, you should note in config those only work on pickups (otherwise weapons can be in a really weird spot after spam replacing models)

Reply:
Will do